app/scripts/models/reliers/relier.js

```diff
@@ -106,6 +106,33 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Set a value based on a value in window.location.search. Only updates
+     * model if parameter exists in window.location.search and passes validation.
+     *
+     * Throws Error mapped to `INVALID_PARAMETER` in Errors object.
+     *
+     * @param {String} sourceName - name of the search parameter
+     * @param {String} destName - name to set in model
+     * @param {Function} validateFunc - validation function, returns `true` or `false`
+     * @param {Errors} Errors - corresponding Errors object
+     * @throws {Error}
+     */
+    _importValidateOptionalSearchParam: function (sourceName, destName, validateFunc, Errors) {
```

Comment:
As an aside, this is probably an opportune moment to push down [_importRequiredSearchParam](https://github.com/mozilla/fxa-content-server/blob/0c84ed2695abf0411c875aa499f61be38e191464/app/scripts/models/reliers/oauth.js#L150) from oauth.js to relier.js (removing the `_` prefix too!)